app/services/foreman/plugin.rb

```diff
@@ -243,7 +243,9 @@ def pending_migrations
       pending_migrations = ActiveRecord::Migrator.new(:up, migration_paths).
         pending_migrations
 
-      pending_migrations.size > 0
+      return false unless pending_migrations.size > 0
+      Rails.logger.warn(
```

Comment:
This generates a lot of information on every new installation with a plugin - i.e. when calling `rake db:migrate` to set up a new database, you receive a huge log message every time `role` is called from a plugin.

    2016-10-05T09:26:16  [app] [W] Pending migrations: CreateHosts, AddAuditsTable, CreateArchitectures, CreateMedia, CreateDomains, CreateSubnets, CreateOperatingsystems, CreateModels, CreateEnvironments, CreateReports, CreatePtables, CreatePuppetclasses, CreateParameters, C
    reateAuthSources, CreateHostgroups, AddTypeToParameter, AddDomainToParameter, ChangeReportFieldTypeToText, ChangeHostBuildDefaultToFalse, AddIndexToHost, AddIndexToParameters, AddIndexToReports, CreateUsers, CreateSessions, AddStatusToReport, CalcExistingReports, CreateLo
    okupKeys, CreateLookupValues, ChangeMysqlReportsColumn, AddFamilyToOs, UpdateFactNamesAndValuesToBin, AddReleasenameToOs, CreateUsergroups, AddOwnerToHosts, AddUbuntuCustomLvmPtable, CreateMediaOperatingsystemsAndMigrateData, MigrateInstallationMediumUri, SimplifyParamete
    rs, UpdateOsMinor, ChangeFamilyInOs, CreateNotices, AddInternalAuth, CreateRoles, AddUserDomainsAndHostgroups, CreateUserFacts, AddEnabledToHosts, AddMyBooleansToUser, CreateMessages, CreateSources, CreateLogs, AddMetricsToReport, ConvertReports, AddOwnedFilterToUser, Cor
    rectMedia, CreateConfigTemplates, CreateTemplateCombinations, CreateOperatingsystemsConfigTemplates, CreateTemplateKinds, CreateOsDefaultTemplates, AddDefaultTemplates, AddSubSystemsToSubnet, CreateProxies, AddProxyToDomain, AddGrubbyTemplate, AddHostGroupDefaults, AddDef
    aultPxeMenuTemplate, CreateProxyFeatures, ChangePuppetmasterColumn, AddManagedToHosts, RevertFaceNamesAndValuesToTextRecords, AddPriorityToParameter, AddMediaPathToMedium, AddImageToHost, RemoveUnusedFieldsFromPuppetClasses, AddTableBookmarks, AddSolarisTemplates, AddSola
    risDisks, AddLookUpKeyIdToPuppetClass, AddSparcInfoToModel, AddAncestryToHostgroup, CreateSettings, UpdateAuditsTable, CreateHostClass, AddSuseTemplates, AddVmDefaultsToHostgroup, AddSubnetAndDomainToHostGroups, RenameChangesToAuditedChanges, AddRemoteAddressToAudits, Ren
    ameParentToAssociation, AddGatewayAndDnsToSubnets, EnsureAllHostnamesAreLowercase, AddFromAndToRangesToSubnets, CreateSubnetDomain, CreateComputeResources, AddUuidAndComputeIdToHost, AddWindowsTemplates, CreateNics, AddPuppetMasterProxyToHostAndHostGroup, UpdateReportFiel
    dToLargeInt, CreateImages, AddCertNameToHost, CreateKeyPairs, AddDnsIdToSubnet, AddImageIdToHost, AddUserComputeResources, AddComputeResourcesBooleanToUser, AddEpelSnippets, ConvertParamsToText, ChangeBookmarkColumnToText, RenameAssociationToAssociated, FixAuditableType, 
    FixAssociatedType, CreateTaxonomies, AddOkHostsBookMark, AddAuditableNameAndAssociatedNameToAudit, AddOsFamilyToMedia, AddOsFamilyToPtable, AddAttributesToComputeResources, AddSomeIndexes, CreateEnvironmentClasses, AddLookupKeysOverrideAndRequired, AddLookupValuesCountToL
    ookupKeys, CacheLookupValuesCount, AddIamRoleToImages, CreateTokens, AddBmcFeatureToProxy, AddTaxonomyIdsToHosts, ChangeDomainCharacterCount, CreateTrends, CreateTrendCounters, UpdateMediaPathLimit, CreateHostgroupClasses, AddLocalbootTemplate, RenameReplyAdressSetting, A
    ddLocationsOrganizations, MigrateHypervisorsToComputeResources, CreateTaxableTaxonomies, AddMyTaxonomyToUser, AddSubscribeToAllHostgroupsToUsers, ChangeSmartVariableLength, RemoveReplayAddressSetting, AddIgnoreTypesToTaxonomy, AddDigestToMessages, AddStiToHosts, AddDigest
    ToSources, RemoveDuplicateSnippets, AddLabelToHostGroup, AddStiToSettings, FixStiHostAuditableType, AddIdToUserHostgroup, AddLocaleToUsers, RemoveDuplicateFactNames, AddUniqueConstraintToFactName, AddUniqueConstraintsToFactValues, ChangeColumnLengths, AddTemplatesToFeatur
    es, RemoveEnvironmentFromHost, AddPublicToKeyPairs, ChangeComputeResourcePasswordToText, AddLdapFilterToAuthSources, DeleteOrphanedRecords, AddKeys, RemoveUnusedRoleFields, FixAuditableType2, AddComposeFlagAndShortNameToFactName, AllowNullValuesOnFactValue, AddTypeToFactN
    ame, ChangeNameIndexOnFactName, UpdateForemanUrl, AddAncestryToFactNames, ExtendUserRole, CreateCachedUserRoles, CalculateCacheForUserRole, AddPxegrubLocalbootTemplate, CreateCachedUsergroupMembers, CreateRealms, RenameSeededTemplates, AddAdminFlagToUsergroup, CreatePermi
    ssions, CreateFilters, CreateFilterings, AddDescriptionToOperatingsystem, AddLdapAvatarSupport, AddUserdataFlagToImages, CreateComputeProfiles, CreateComputeAttributes, AddComputeProfileToHostgroup, AddForeignKeysToFiltersAndFilterings, AddPasswordToImages, AddDefaultOrga
    nizationIdToUsers, AddDefaultLocationIdToUsers, AddAncestryToTaxonomies, ChangeLabelToTitle, MigratePermissions, AddTaxonomySearchesToFilter, AddProvisionMethodToHosts, AddCounterCaches, FixPuppetclassCounters, ChangeHostManagedDefaultToFalse, AddServerTypeToAuthSource, A
    ddGroupsBaseToAuthSource, AddLowerloginToUsers, CreateConfigGroups, CreateConfigGroupClasses, CreateHostConfigGroups, CreateExternalUsergroups, RemoveSubscribeToAllHostgroupsFromUsers, AddCountersToConfigGroups, AddConfigGroupCounterDefaults, CleanupUsersProperties, Remov
    eUserJoinTables, AddHiddenValueToParameter, FixIntegerSettings, AddBootModeToSubnet, AddLockedAndDefaultAndVendorToConfigTemplate, RemoveArchitectureDefault, ExtractNicAttributes, AddAttributesToNicBase, AddManagedToNics, ChangeAuditableNameColumnToText, AddPrimaryInterfa
    ceToHosts, AddIpSuggestionToSubnets, AddUniqueIndexToParameter, AddMergeOverridesAndAvoidDuplicatesToLookupKey, FixCounters, AddRequestUuidToAudits, ConvertIpamToString, RemoveSignoSetting, ChangeNilAdminUsersToFalse, AddBondAttributesToNicBase, MoveHostNicsToInterfaces, 
    AddPasswordHashToOperatingsystem, AddGrubPasswordToHosts, AddGrubPasswordToHostgroup, AddShouldUsePuppetDefaultToLookupValueAndKey, AddTitleToOs, CreateWidgets, CreateMailNotifications, CreateUserMailNotifications, AddMailEnabledToUser, AddSubscriptionTypeToMailNotificati
    ons, RemoveFailedReportSetting, RenameSubnetNameToUnique, AddTimezoneToUser, RenameHostsCountColumn, FixPuppetclassTotalHosts, AddTypeToParameterIndex, AddMailQueryToUserMailNotification, AddQueryableToMailNotification, RemoveAuditUserFk, RenameSmartProxyAuthRelatedSettin
    gs, MakeTemplatesDefault, AddComputeAttributesToNics, AddDescriptionToTaxonomies, ActuallyRenameSmartProxyAuthRelatedSettings, SetPtableLayoutToText, AddRangeToTrendCounters, MoveDescriptionFieldsToText, AddDefaultWidgets, ChangeDefaultSubnetBootMode, MigrateWebsocketsSet
    ting, AddIndexToLogsSourceId, ChangeOsDefaultPasswordHash, CopyUnmanagedHostsToInterfaces, AddTypeToConfigTemplate, MigratePtablesToTemplates, AddUsergroupSyncToAuthSources, RemoveTotalHostsAudits, RenameConfigTemplateToProvisioningTemplate, RemoveDuplicateTokens, ChangeU
    sergroupNameToBeRequired, RemoveCounterCacheAudits, FixTemplateSnippetFlag, AssignPtablesToTaxonomies, AddStiToLookupKeys, RenameTaxonomyIgnoredTypeToProvisioningTemplates, CreateHostStatus, RenamePuppetclassCountersForLkSti, RemoveUnusedFieldsFromHosts, ChangeReportedAt,
     AddTypeToReports, AddFullNameToSetting, AddUniqueToOperatingsystemsTitle, RemoveIsParam, RemoveNewFromComputeAttributes, RemoveChefProxy, ChangeBookmarkReportController, ChangeReportPermissions, AddMergeDefaultToLookupKey, AddLookupValueMatchToHostAndHostgroup, UpdateHos
    tStatusStatusFieldInt, DropPtables, RenameDomainHostCount, AddHiddenValueToLookupKey, AddHostsCountToHostgroup, AddHostStatusHostIdIndex, DeleteBootableInterface, AddTypeToMailNotification, RemoveSpacesFromSmartVariableKey, AddExpiredLogsToSmartProxy, AddDefaultValueForBo
    okmarkPublicField, AddSubnetDomainRelationConstraints, RemoveDefaultUserRole, RenameAnonymousRole, UpdateParamsPriority, RemoveHostgroupsCountFromPuppetclasses, RemovePermissionsFromRoles, AddViewParamsToFiltersWithEdit, DowncaseDisplayTypes, AddStiToSubnets, AddIpv6ToHos
    ts, AddIpv6Subnet, DivideLookupKeyPermissions, LimitOsDescription, RemoveNilFromMergeOverride, RemoveHostCounterCache, RemovePuppetCounters, SetRoleBuiltinDefault, UnifyPermissions, ChangeTemplatesTypeDefault, ChangeTemplateTaxableTaxonomiesType, ChangeTemplateAuditsType,
     AddPxeLoaderToHost, RenameLookupValueUsePuppetDefault, AddDescriptionToUser, AddPxeLoaderToHostgroup, AddDescriptionToRole, ResetOverrideParams, AddOverrideFlagToFilter, RenameLookupKeyUsePuppetDefault, UpdateParameterPriorities, ChangeFactNameToPuppetFactName, ChangeWid
    getNames